src/app/repositories/firebase/schema_firebase_repository.py

```diff
@@ -154,3 +154,20 @@ def get_schema_metadata_collection(self, survey_id: str) -> list[SchemaMetadata]
             schema_metadata_list.append(metadata)
 
         return schema_metadata_list
+
+    def get_list_survey_id(self) -> list[str]:
+        """
+        Gets the list of unique Survey IDs from the 'schemas' collection in Firestore.
+        """
+
+        list_survey_id = []
+        returned_schema = self.schemas_collection.order_by(
+            "survey_id", direction=firestore.Query.ASCENDING
+        ).stream()
+
+        for schema in returned_schema:
+            survey_id = schema.to_dict()["survey_id"]
+            if survey_id not in list_survey_id:
```

Comment:
A bit of suggestion to enhance performance. Since the list is sorted, you may just check whether the survey_id is the same as the previous one that is appended.

Reply:
@IsaiahLai111 Good point, just changed the logic to check only for the latest entry, please have a look, thanks.

Reply:
It's nice. Thank you